PyYAML fails to install on Python 3.7

Python 3.7 has been released recently, and on OS-X, Travis uses the latest Python version, which automatically became 3.7 when it was released. On Python 3.7, PyYAML fails to install, with a long list of errors:

```
  Running setup.py clean for PyYAML
Successfully built ply
Failed to build PyYAML
Installing collected packages: six, ply, PyYAML, mock
  Running setup.py install for PyYAML ... error
    Complete output from command /Users/travis/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/pip-build-7r95wakx/PyYAML/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/pip-3d2jbxkh-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/travis/venv/bin/../include/site/python3.7/PyYAML:
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.13-x86_64-3.7
    creating build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/__init__.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/composer.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/constructor.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/cyaml.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/dumper.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/emitter.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/error.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/events.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/loader.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/nodes.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/parser.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/reader.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/representer.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/resolver.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/scanner.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/serializer.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    copying lib3/yaml/tokens.py -> build/lib.macosx-10.13-x86_64-3.7/yaml
    running build_ext
    creating build/temp.macosx-10.13-x86_64-3.7
    checking if libyaml is compilable
    clang -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/opt/openssl/include -I/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c build/temp.macosx-10.13-x86_64-3.7/check_libyaml.c -o build/temp.macosx-10.13-x86_64-3.7/check_libyaml.o
    checking if libyaml is linkable
    clang build/temp.macosx-10.13-x86_64-3.7/check_libyaml.o -lyaml -o build/temp.macosx-10.13-x86_64-3.7/check_libyaml
    building '_yaml' extension
    creating build/temp.macosx-10.13-x86_64-3.7/ext
    clang -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/opt/openssl/include -I/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c ext/_yaml.c -o build/temp.macosx-10.13-x86_64-3.7/ext/_yaml.o
    In file included from ext/_yaml.c:271:
    ext/_yaml.h:10:9: warning: 'PyString_CheckExact' macro redefined [-Wmacro-redefined]
    #define PyString_CheckExact PyBytes_CheckExact
            ^
    ext/_yaml.c:139:11: note: previous definition is here
      #define PyString_CheckExact          PyUnicode_CheckExact
              ^
    ext/_yaml.c:1410:17: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
      __pyx_v_value = yaml_get_version_string();
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
    ext/_yaml.c:2577:52: warning: incompatible pointer types passing 'int (void *, char *, size_t, size_t *)' (aka 'int (void *, char *, unsigned long, unsigned long *)') to parameter of type 'yaml_read_handler_t *' (aka 'int (*)(void *, unsigned char *, unsigned long, unsigned long *)') [-Wincompatible-pointer-types]
        yaml_parser_set_input((&__pyx_v_self->parser), __pyx_f_5_yaml_input_handler, ((void *)__pyx_v_self));
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/include/yaml.h:1368:30: note: passing argument to parameter 'handler' here
            yaml_read_handler_t *handler, void *data);
                                 ^
    ext/_yaml.c:2818:59: warning: passing 'char *' to parameter of type 'const unsigned char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        yaml_parser_set_input_string((&__pyx_v_self->parser), PyString_AS_STRING(__pyx_v_stream), PyString_GET_SIZE(__pyx_v_stream));
                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ext/_yaml.h:11:29: note: expanded from macro 'PyString_AS_STRING'
    #define PyString_AS_STRING  PyBytes_AS_STRING
                                ^
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/bytesobject.h:87:31: note: expanded from macro 'PyBytes_AS_STRING'
    #define PyBytes_AS_STRING(op) (assert(PyBytes_Check(op)), \
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/include/yaml.h:1342:30: note: passing argument to parameter 'input' here
            const unsigned char *input, size_t size);
                                 ^
    ext/_yaml.c:4572:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag_directive.handle); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 417, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:4584:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag_directive.prefix); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 418, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:5444:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.alias.value); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 448, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:5518:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.anchor.value); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 451, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:5592:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag.handle); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 454, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:5604:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_FromString(__pyx_v_token->data.tag.suffix); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 455, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:5716:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_DecodeUTF8(__pyx_v_token->data.scalar.value, __pyx_v_token->data.scalar.length, ((char *)"strict")); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 460, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:1299:17: note: passing argument to parameter 'string' here
        const char *string,         /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:7424:42: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
            __pyx_t_4 = PyUnicode_FromString(__pyx_v_tag_directive->handle); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 574, __pyx_L1_error)
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:7436:42: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
            __pyx_t_4 = PyUnicode_FromString(__pyx_v_tag_directive->prefix); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 575, __pyx_L1_error)
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:7655:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.alias.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 586, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:7749:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.scalar.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 591, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:7790:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.scalar.tag); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 594, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:7811:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_4 = PyUnicode_DecodeUTF8(__pyx_v_event->data.scalar.value, __pyx_v_event->data.scalar.length, ((char *)"strict")); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 595, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:1299:17: note: passing argument to parameter 'string' here
        const char *string,         /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:8179:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.sequence_start.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 620, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:8220:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.sequence_start.tag); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 623, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:8449:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.mapping_start.anchor); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 637, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:8490:40: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_4 = PyUnicode_FromString(__pyx_v_event->data.mapping_start.tag); if (unlikely(!__pyx_t_4)) __PYX_ERR(0, 640, __pyx_L1_error)
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:10094:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_3 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.alias.anchor); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 734, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:10378:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_6 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.scalar.anchor); if (unlikely(!__pyx_t_6)) __PYX_ERR(0, 750, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:10434:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_6 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.sequence_start.anchor); if (unlikely(!__pyx_t_6)) __PYX_ERR(0, 753, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:10490:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_6 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.mapping_start.anchor); if (unlikely(!__pyx_t_6)) __PYX_ERR(0, 756, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:11109:36: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
      __pyx_t_2 = PyUnicode_DecodeUTF8(__pyx_v_self->parsed_event.data.scalar.value, __pyx_v_self->parsed_event.data.scalar.length, ((char *)"strict")); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 791, __pyx_L1_error)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:1299:17: note: passing argument to parameter 'string' here
        const char *string,         /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:11310:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.scalar.tag); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 804, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:11841:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_3 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.sequence_start.tag); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 837, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:12435:38: warning: passing 'yaml_char_t *' (aka 'unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_3 = PyUnicode_FromString(__pyx_v_self->parsed_event.data.mapping_start.tag); if (unlikely(!__pyx_t_3)) __PYX_ERR(0, 876, __pyx_L1_error)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/unicodeobject.h:703:17: note: passing argument to parameter 'u' here
        const char *u              /* UTF-8 encoded string */
                    ^
    ext/_yaml.c:13630:53: warning: incompatible pointer types passing 'int (void *, char *, size_t)' (aka 'int (void *, char *, unsigned long)') to parameter of type 'yaml_write_handler_t *' (aka 'int (*)(void *, unsigned char *, unsigned long)') [-Wincompatible-pointer-types]
      yaml_emitter_set_output((&__pyx_v_self->emitter), __pyx_f_5_yaml_output_handler, ((void *)__pyx_v_self));
                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /usr/local/include/yaml.h:1830:31: note: passing argument to parameter 'handler' here
            yaml_write_handler_t *handler, void *data);
                                  ^
    ext/_yaml.c:15002:44: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types with different sign [-Wpointer-sign]
            __pyx_v_tag_directives_end->handle = PyString_AS_STRING(__pyx_v_handle);
                                               ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ext/_yaml.c:15117:44: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types with different sign [-Wpointer-sign]
            __pyx_v_tag_directives_end->prefix = PyString_AS_STRING(__pyx_v_prefix);
                                               ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ext/_yaml.c:15454:62: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = ((yaml_alias_event_initialize(__pyx_v_event, __pyx_v_anchor) == 0) != 0);
                                                                 ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:553:63: note: passing argument to parameter 'anchor' here
    yaml_alias_event_initialize(yaml_event_t *event, yaml_char_t *anchor);
                                                                  ^
    ext/_yaml.c:16169:63: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = ((yaml_scalar_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0) != 0);
                                                                  ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:579:22: note: passing argument to parameter 'anchor' here
            yaml_char_t *anchor, yaml_char_t *tag,
                         ^
    ext/_yaml.c:16169:79: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = ((yaml_scalar_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0) != 0);
                                                                                  ^~~~~~~~~~~
    /usr/local/include/yaml.h:579:43: note: passing argument to parameter 'tag' here
            yaml_char_t *anchor, yaml_char_t *tag,
                                              ^
    ext/_yaml.c:16169:92: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = ((yaml_scalar_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0) != 0);
                                                                                               ^~~~~~~~~~~~~
    /usr/local/include/yaml.h:580:22: note: passing argument to parameter 'value' here
            yaml_char_t *value, int length,
                         ^
    ext/_yaml.c:16603:71: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_3 = ((yaml_sequence_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0) != 0);
                                                                          ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:602:22: note: passing argument to parameter 'anchor' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                         ^
    ext/_yaml.c:16603:87: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_3 = ((yaml_sequence_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0) != 0);
                                                                                          ^~~~~~~~~~~
    /usr/local/include/yaml.h:602:43: note: passing argument to parameter 'tag' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                              ^
    ext/_yaml.c:17037:70: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = ((yaml_mapping_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0) != 0);
                                                                         ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:634:22: note: passing argument to parameter 'anchor' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                         ^
    ext/_yaml.c:17037:86: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = ((yaml_mapping_start_event_initialize(__pyx_v_event, __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0) != 0);
                                                                                         ^~~~~~~~~~~
    /usr/local/include/yaml.h:634:43: note: passing argument to parameter 'tag' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                              ^
    ext/_yaml.c:18516:42: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_v_tag_directives_end->handle = PyString_AS_STRING(__pyx_v_handle);
                                             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ext/_yaml.c:18631:42: warning: assigning to 'yaml_char_t *' (aka 'unsigned char *') from 'char *' converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_v_tag_directives_end->prefix = PyString_AS_STRING(__pyx_v_prefix);
                                             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ext/_yaml.c:19491:65: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
        __pyx_t_2 = ((yaml_alias_event_initialize((&__pyx_v_event), __pyx_v_anchor) == 0) != 0);
                                                                    ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:553:63: note: passing argument to parameter 'anchor' here
    yaml_alias_event_initialize(yaml_event_t *event, yaml_char_t *anchor);
                                                                  ^
    ext/_yaml.c:20240:68: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_2 = ((yaml_scalar_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0) != 0);
                                                                       ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:579:22: note: passing argument to parameter 'anchor' here
            yaml_char_t *anchor, yaml_char_t *tag,
                         ^
    ext/_yaml.c:20240:84: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_2 = ((yaml_scalar_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0) != 0);
                                                                                       ^~~~~~~~~~~
    /usr/local/include/yaml.h:579:43: note: passing argument to parameter 'tag' here
            yaml_char_t *anchor, yaml_char_t *tag,
                                              ^
    ext/_yaml.c:20240:97: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_2 = ((yaml_scalar_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_value, __pyx_v_length, __pyx_v_plain_implicit, __pyx_v_quoted_implicit, __pyx_v_scalar_style) == 0) != 0);
                                                                                                    ^~~~~~~~~~~~~
    /usr/local/include/yaml.h:580:22: note: passing argument to parameter 'value' here
            yaml_char_t *value, int length,
                         ^
    ext/_yaml.c:20605:76: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_2 = ((yaml_sequence_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0) != 0);
                                                                               ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:602:22: note: passing argument to parameter 'anchor' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                         ^
    ext/_yaml.c:20605:92: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_2 = ((yaml_sequence_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_sequence_style) == 0) != 0);
                                                                                               ^~~~~~~~~~~
    /usr/local/include/yaml.h:602:43: note: passing argument to parameter 'tag' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                              ^
    ext/_yaml.c:21113:75: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_2 = ((yaml_mapping_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0) != 0);
                                                                              ^~~~~~~~~~~~~~
    /usr/local/include/yaml.h:634:22: note: passing argument to parameter 'anchor' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                         ^
    ext/_yaml.c:21113:91: warning: passing 'char *' to parameter of type 'yaml_char_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
          __pyx_t_2 = ((yaml_mapping_start_event_initialize((&__pyx_v_event), __pyx_v_anchor, __pyx_v_tag, __pyx_v_implicit, __pyx_v_mapping_style) == 0) != 0);
                                                                                              ^~~~~~~~~~~
    /usr/local/include/yaml.h:634:43: note: passing argument to parameter 'tag' here
            yaml_char_t *anchor, yaml_char_t *tag, int implicit,
                                              ^
    ext/_yaml.c:24143:21: error: no member named 'exc_type' in 'struct _ts'
        *type = tstate->exc_type;
                ~~~~~~  ^
    ext/_yaml.c:24144:22: error: no member named 'exc_value' in 'struct _ts'; did you mean 'curexc_value'?
        *value = tstate->exc_value;
                         ^~~~~~~~~
                         curexc_value
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
        PyObject *curexc_value;
                  ^
    ext/_yaml.c:24145:19: error: no member named 'exc_traceback' in 'struct _ts'; did you mean 'curexc_traceback'?
        *tb = tstate->exc_traceback;
                      ^~~~~~~~~~~~~
                      curexc_traceback
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared here
        PyObject *curexc_traceback;
                  ^
    ext/_yaml.c:24152:24: error: no member named 'exc_type' in 'struct _ts'
        tmp_type = tstate->exc_type;
                   ~~~~~~  ^
    ext/_yaml.c:24153:25: error: no member named 'exc_value' in 'struct _ts'; did you mean 'curexc_value'?
        tmp_value = tstate->exc_value;
                            ^~~~~~~~~
                            curexc_value
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
        PyObject *curexc_value;
                  ^
    ext/_yaml.c:24154:22: error: no member named 'exc_traceback' in 'struct _ts'; did you mean 'curexc_traceback'?
        tmp_tb = tstate->exc_traceback;
                         ^~~~~~~~~~~~~
                         curexc_traceback
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared here
        PyObject *curexc_traceback;
                  ^
    ext/_yaml.c:24155:13: error: no member named 'exc_type' in 'struct _ts'
        tstate->exc_type = type;
        ~~~~~~  ^
    ext/_yaml.c:24156:13: error: no member named 'exc_value' in 'struct _ts'; did you mean 'curexc_value'?
        tstate->exc_value = value;
                ^~~~~~~~~
                curexc_value
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
        PyObject *curexc_value;
                  ^
    ext/_yaml.c:24157:13: error: no member named 'exc_traceback' in 'struct _ts'; did you mean 'curexc_traceback'?
        tstate->exc_traceback = tb;
                ^~~~~~~~~~~~~
                curexc_traceback
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared here
        PyObject *curexc_traceback;
                  ^
    ext/_yaml.c:24212:24: error: no member named 'exc_type' in 'struct _ts'
        tmp_type = tstate->exc_type;
                   ~~~~~~  ^
    ext/_yaml.c:24213:25: error: no member named 'exc_value' in 'struct _ts'; did you mean 'curexc_value'?
        tmp_value = tstate->exc_value;
                            ^~~~~~~~~
                            curexc_value
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
        PyObject *curexc_value;
                  ^
    ext/_yaml.c:24214:22: error: no member named 'exc_traceback' in 'struct _ts'; did you mean 'curexc_traceback'?
        tmp_tb = tstate->exc_traceback;
                         ^~~~~~~~~~~~~
                         curexc_traceback
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared here
        PyObject *curexc_traceback;
                  ^
    ext/_yaml.c:24215:13: error: no member named 'exc_type' in 'struct _ts'
        tstate->exc_type = local_type;
        ~~~~~~  ^
    ext/_yaml.c:24216:13: error: no member named 'exc_value' in 'struct _ts'; did you mean 'curexc_value'?
        tstate->exc_value = local_value;
                ^~~~~~~~~
                curexc_value
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
        PyObject *curexc_value;
                  ^
    ext/_yaml.c:24217:13: error: no member named 'exc_traceback' in 'struct _ts'; did you mean 'curexc_traceback'?
        tstate->exc_traceback = local_tb;
                ^~~~~~~~~~~~~
                curexc_traceback
    /usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared here
        PyObject *curexc_traceback;
                  ^
    50 warnings and 15 errors generated.
    error: command 'clang' failed with exit status 1
    
    ----------------------------------------
Command "/Users/travis/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/pip-build-7r95wakx/PyYAML/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/pip-3d2jbxkh-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/travis/venv/bin/../include/site/python3.7/PyYAML" failed with error code 1 in /private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/pip-build-7r95wakx/PyYAML/
```

This issue seems to be the same as PyYAML issue 126: https://github.com/yaml/pyyaml/issues/126, which has been solved in PyYAML 3.13.

--- a/.travis.yml
+++ b/.travis.yml
@@ -50,16 +50,16 @@ matrix:
       python: "2.7"
       env:
         - PACKAGE_LEVEL=latest
-    - os: linux
-      language: python
-      python: "3.4"
-      env:
-        - PACKAGE_LEVEL=minimum
 #    - os: linux
 #      language: python
 #      python: "3.4"
 #      env:
-#        - PACKAGE_LEVEL=latest
+#        - PACKAGE_LEVEL=minimum
+    - os: linux
+      language: python
+      python: "3.4"
+      env:
+        - PACKAGE_LEVEL=latest
 #    - os: linux
 #      language: python
 #      python: "3.5"
@@ -75,11 +75,23 @@ matrix:
 #      python: "3.6"
 #      env:
 #        - PACKAGE_LEVEL=minimum
+#    - os: linux
+#      language: python
+#      python: "3.6"
+#      env:
+#        - PACKAGE_LEVEL=latest
     - os: linux
       language: python
-      python: "3.6"
+      python: "3.7"
+      dist: xenial
       env:
-        - PACKAGE_LEVEL=latest
+        - PACKAGE_LEVEL=minimum
+#    - os: linux
+#      language: python
+#      python: "3.7"
+#      dist: xenial
+#      env:
+#        - PACKAGE_LEVEL=latest
 # Note: pywbem does not install on pypy, because M2Crypto does not install.
 #    - os: linux
 #      language: python

--- a/PKG-INFO
+++ b/PKG-INFO
@@ -1,6 +1,6 @@
 Metadata-Version: 1.2
 Name: pywbem
-Version: 0.12.1.dev59
+Version: 0.12.1.dev70
 Summary: pywbem - A WBEM client
 Home-page: http://pywbem.github.io/pywbem/
 Author: Tim Potter
@@ -44,5 +44,6 @@ Classifier: Programming Language :: Python :: 3
 Classifier: Programming Language :: Python :: 3.4
 Classifier: Programming Language :: Python :: 3.5
 Classifier: Programming Language :: Python :: 3.6
+Classifier: Programming Language :: Python :: 3.7
 Classifier: Topic :: Software Development :: Libraries :: Python Modules
 Classifier: Topic :: System :: Systems Administration

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -26,9 +26,15 @@ environment:
 #    - PYTHON_VERSION: 3.6
 #      PYTHON_ARCH: 32
 #      PYTHON_HOME: C:\Python36
-    - PYTHON_VERSION: 3.6
+#    - PYTHON_VERSION: 3.6
+#      PYTHON_ARCH: 64
+#      PYTHON_HOME: C:\Python36-x64
+#    - PYTHON_VERSION: 3.7
+#      PYTHON_ARCH: 32
+#      PYTHON_HOME: C:\Python37
+    - PYTHON_VERSION: 3.7
       PYTHON_ARCH: 64
-      PYTHON_HOME: C:\Python36-x64
+      PYTHON_HOME: C:\Python37-x64
 
 configuration:
 # These values will become the values of the PACKAGE_LEVEL env.var.

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -16,8 +16,8 @@ pytest>=3.3.0; python_version > '2.6'
 # testfixtures 6.0.0 no longer supports py26 and fails on py26 with syntax error
 testfixtures>=4.3.3,<6.0.0
 httpretty>=0.8.14,<0.9.1; python_version == '2.6'
-httpretty>=0.8.14; python_version > '2.6'
-lxml>=4.0.0
+httpretty>=0.9.5; python_version > '2.6'
+lxml>=4.2.4
 requests>=2.12.4
 decorator>=4.0.11
 yamlordereddictloader>=0.4.0

--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -238,6 +238,8 @@ This version contains all fixes up to pywbem 0.12.4.
   according to DSP0004, return equal WBEM URI strings.
   See issue #1323.
 
+* Added support for Python 3.7, which was released 2018-06-27.
+
 **Cleanup:**
 
 * Added connection information to all pywbem exceptions. This is done via a
@@ -265,6 +267,10 @@ This version contains all fixes up to pywbem 0.12.4.
   that creates the pywbem_mock for any tests that depend on classes like
   CIM_Namespace, CIM_ObjectManager existing in the mocked server. See issue
   #1250
+
+* Needed to upgrade PyYAML version from >=3.12 to >=3.13 due to an issue
+  in PyYAML on Python 3.7, that was fixed in PyYAML 3.13.
+  See issue #1337.
 
 **Known issues:**
 

--- a/docs/intro.rst
+++ b/docs/intro.rst
@@ -98,7 +98,7 @@ Pywbem is supported in these environments:
 * Operating Systems: Linux, Windows (native, and with UNIX-like environments),
   OS-X
 
-* Python: 2.6, 2.7, 3.4, 3.5, 3.6 and higher
+* Python: 2.6, 2.7, 3.4, and higher
 
 * WBEM servers: Any WBEM server that conforms to the DMTF specifications listed
   in :ref:`Standards conformance`. WBEM servers supporting older versions of

--- a/minimum-constraints.txt
+++ b/minimum-constraints.txt
@@ -47,7 +47,7 @@ wheel===0.29.0
 pbr===1.10.0
 six===1.10.0
 ply===3.10
-PyYAML===3.12
+PyYAML===3.13
 M2Crypto===0.30.1
 ordereddict===1.1
 
@@ -63,8 +63,9 @@ pytest===3.0.7; python_version == '2.6'
 pytest===3.3.0; python_version > '2.6'
 testfixtures===4.3.3
 mock===2.0.0
-httpretty===0.8.14
-lxml===4.0.0
+httpretty===0.8.14; python_version == '2.6'
+httpretty===0.9.5; python_version > '2.6'
+lxml===4.2.4
 requests===2.12.4
 decorator===4.0.11
 yamlordereddictloader===0.4.0
@@ -148,7 +149,7 @@ py===1.4.32; python_version == '2.6'
 py===1.5.1; python_version > '2.6'
 Pygments===2.1.3
 pytz===2016.10
-pyzmq===16.0.2
+pyzmq===16.0.4
 qtconsole===4.2.1
 requests===2.12.4
 requests-toolbelt===0.7.0

--- a/pywbem/_nocasedict.py
+++ b/pywbem/_nocasedict.py
@@ -186,9 +186,9 @@ class NocaseDict(object):
                 "%s (1 allowed)" % len(args))
 
         # Step 2: Add any keyword arguments
-        if len(kwargs) > 1 and sys.version_info[0:2] <= (3, 6):
+        if len(kwargs) > 1 and sys.version_info[0:2] < (3, 7):
             warnings.warn("Initializing a pywbem.NocaseDict object from "
-                          "keyword arguments before Python 3.6 will not "
+                          "keyword arguments before Python 3.7 will not "
                           "preserve order of items",
                           UserWarning,
                           stacklevel=_stacklevel_above_module(__name__))

--- a/pywbem/cim_types.py
+++ b/pywbem/cim_types.py
@@ -626,6 +626,16 @@ class CIMInt(CIMType, _Longint):
     maxvalue = None
 
     def __new__(cls, *args, **kwargs):
+
+        # Python 3.7 removed support for passing the value for int() as a
+        # keyword argument named 'x'. It now needs to be passed as a positional
+        # argument. The testclient test case definitions rely on a keyword
+        # argument, so we now transform the keyword arg into a positional
+        # arg.
+        if 'x' in kwargs:
+            args = list(*args)  # args is passed as a tuple
+            args.append(kwargs.pop('x'))
+
         value = _Longint(*args, **kwargs)
         if ENFORCE_INTEGER_RANGE:
             if value > cls.maxvalue or value < cls.minvalue:

--- a/requirements.txt
+++ b/requirements.txt
@@ -12,7 +12,7 @@
 pbr>=1.10.0
 six>=1.10.0
 ply>=3.10
-PyYAML>=3.12  # yaml package
+PyYAML>=3.13  # yaml package
 # On Windows, M2Crypto must be installed via pywbem_os_setup.bat
 M2Crypto>=0.30.1; python_version < '3.0' and sys_platform != 'win32'
 ordereddict>=1.1; python_version == '2.6'

--- a/rtd-requirements.txt
+++ b/rtd-requirements.txt
@@ -6,7 +6,7 @@
 pbr>=1.10.0
 six>=1.10.0
 ply>=3.8
-PyYAML>=3.12
+PyYAML>=3.13
 # M2Crypto>=0.30.1  # we cannot install M2Crypto because RTD does not have Swig
 Sphinx>=1.4.9
 sphinx-git>=10.0.0

--- a/setup.cfg
+++ b/setup.cfg
@@ -41,6 +41,7 @@ classifier =
     Programming Language :: Python :: 3.4
     Programming Language :: Python :: 3.5
     Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
     Topic :: Software Development :: Libraries :: Python Modules
     Topic :: System :: Systems Administration
 

--- a/testsuite/test_nocasedict.py
+++ b/testsuite/test_nocasedict.py
@@ -4,6 +4,7 @@ Test the _nocasedict module.
 
 from __future__ import absolute_import
 
+import sys
 import re
 import six
 try:
@@ -149,7 +150,7 @@ TESTCASES_NOCASEDICT_INIT = [
             exp_dict=OrderedDict([('Dog', 'Cat'), ('Budgie', 'Fish')]),
             verify_order=False,
         ),
-        None, UserWarning, True
+        None, UserWarning if sys.version_info[0:2] < (3, 7) else None, True
     ),
     (
         "Dict from list as positional arg and keyword args",

